app/src/verifier.h

```diff
@@ -173,36 +178,172 @@ namespace scitt::verifier
         throw VerificationError(
           "Resolved verification method public key does not match signing key");
       }
+
+      return payload;
+    }
+
+    std::span<uint8_t> process_signed_statement_with_didattestedsvc_issuer(
+      const cose::ProtectedHeader& phdr,
+      const Configuration& configuration,
+      const std::vector<uint8_t>& data)
+    {
+      check_is_accepted_algorithm(phdr, configuration);
+
+      if (!phdr.tss_map.cose_key.has_value())
+      {
+        throw VerificationError(fmt::format(
+          "Signed statement protected header {} must contain a COSE key",
+          cose::COSE_HEADER_PARAM_TSS));
+      }
+
+      if (!phdr.kid.has_value())
+      {
+        throw VerificationError("Protected header KID is missing");
+      }
+      auto cose_key_thumb = phdr.tss_map.cose_key->to_sha256_thumb();
+      std::vector<uint8_t> kid_bytes(phdr.kid->begin(), phdr.kid->end());
+      if (!std::equal(
+            cose_key_thumb.begin(),
+            cose_key_thumb.end(),
+            kid_bytes.begin(),
+            kid_bytes.end()))
+      {
+        throw VerificationError("COSE key thumbprint does not match KID");
+      }
+
+      PublicKey key = phdr.tss_map.cose_key->to_public_key();
+
+      // First verify the signature to then trust the structure integrity
+      // and the attestation contained in the protected header.
+      std::span<uint8_t> payload;
+      try
+      {
+        // ignore unknown critical headers due to the presence of a custom TSS
+        // header
+        payload = cose::verify(data, key, true);
+      }
+      catch (const cose::COSESignatureValidationError& e)
+      {
+        throw VerificationError(fmt::format(
+          "Failed to validate cose signature using the COSE key: {}",
+          e.what()));
+      }
+
+      // Verify the attestation report contained in the protected header
+      // against the AMD certificate chain contained in “snp_endorsements”.
+      // see
+      // https://github.com/microsoft/CCF/blob/afc7ef5eca00d413474de47f91a1827f16618de6/src/js/extensions/snp_attestation.cpp#L35
+      ccf::QuoteInfo quote_info = {};
+      quote_info.format = ccf::QuoteFormat::amd_sev_snp_v1;
+      quote_info.quote = phdr.tss_map.attestation.value();
+      quote_info.endorsements = phdr.tss_map.snp_endorsements.value();
+
+      ccf::pal::PlatformAttestationMeasurement measurement = {};
+      ccf::pal::PlatformAttestationReportData report_data = {};
+      std::optional<ccf::pal::UVMEndorsements> parsed_uvm_endorsements;
+
+      try
+      {
+        ccf::pal::verify_snp_attestation_report(
+          quote_info, measurement, report_data);
```

Comment:
This extracts the measurement from the attestation, but I think we want to go further than this and call `ccf::pal::verify_uvm_endorsements_descriptor()` with the endorsements to extract a descriptor `(did, feed, svn) atm, CWT claims later`, which can be exposed to the policy.
It's fine to expose the raw measurement too (why not), but policies that check it directly will be quite brittle.